File: Lit-JS/board.js
import {LitElement, html, css} from 'https://cdn.jsdelivr.net/gh/lit/dist@2.2.1/core/lit-core.min.js'
import {boardStyles} from './css.js'
import "./square.js";

export class Board extends LitElement {
    constructor() {
        super();
        this.squares = Array(9).fill("");
        this.onClick = null
    }

    static properties = {
        squares: { type: Array },
        onClick: { attribute: false }
    }

    static styles = boardStyles

    renderSquare(i) {
        return html `
            <square-world
                value = "${this.squares[i]}"
                .onClick=${() => this.onClick(i)}>
            </square-world>`;
    }

    render() {
        return html `
            <div>
                <div class="board-row">
                    ${this.renderSquare(0)}
                    ${this.renderSquare(1)}
                    ${this.renderSquare(2)}
                </div>
                <div class="board-row">
                    ${this.renderSquare(3)}
                    ${this.renderSquare(4)}
                    ${this.renderSquare(5)}
                </div>
                <div class="board-row">
                    ${this.renderSquare(6)}
                    ${this.renderSquare(7)}
                    ${this.renderSquare(8)}
                </div>
            </div>
        `
    }
}

customElements.define( 'board-world', Board );

File: React-Fable/src/Board.fs
module Board

open Feliz
open Square

[<ReactComponent>]
let Board (squares:string list) (onClick:int -> unit) =
    let renderSquare (i) =
        Square (squares |> List.item i) (fun _ -> onClick i)

    Html.div [
        Html.div [
            prop.className "board-row"
            prop.children [
                renderSquare 0
                renderSquare 1
                renderSquare 2
            ]
        ]
        Html.div [
            prop.className "board-row"
            prop.children [
                renderSquare 3
                renderSquare 4
                renderSquare 5
            ]
        ]
        Html.div [
            prop.className "board-row"
            prop.children [
                renderSquare 6
                renderSquare 7
                renderSquare 8
            ]
        ]
    ]